<!-- 
    test_with_virtual_environment.launch
-->

<launch>
<!-- Arguments Start -->
  <arg name="map"            default="auto_drive.bt"/>
  <arg name="bagfile"        default="realtime_mapping.bag"/>
  <arg name="playBagFile"    default="false"/>
<!-- Arguments End -->

<!-- Nodes Start -->
  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server" output="screen" args="$(find atypical_driving)/worlds/$(arg map)" unless="$(arg playBagFile)">
    <param name="resolution"                 value="0.1" />
    <param name="frame_id"                   value="map" type="string"/>
    <param name="sensor_model/max_range"     value="-1" />
    <param name="height_map"                 value="false"/>
    <param name="color/r"                    value="0.2" />
    <param name="color/g"                    value="0.2" />
    <param name="color/b"                    value="0.2" />
    <param name="color/a"                    value="0.2" />

    <remap from="octomap_full" to="/atypical_planning_test/global_map"/>
  </node>

  <node pkg="rosbag" type="play" name="rosbag_record_diag" args="$(find atypical_driving)/worlds/$(arg bagfile)" if="$(arg playBagFile)">
    <remap from="octomap_binary" to="/atypical_planning_test/local_map"/>
  </node> 

  <node pkg="atypical_driving" type="integration_server" name="atypical_planning_test" output="screen">
    <param name="global_planner/horizon" value="15" />
    <param name="global_planner/car_width" value="2" />
    <param name="global_planner/car_z_min" value="0.5" />
    <param name="global_planner/car_z_max" value="1.0" />
    <param name="global_planner/car_speed" value="1.0" />
    <param name="global_planner/road_width" value="4.0" />
    <param name="global_planner/world_x_min" value="-10" />
    <param name="global_planner/world_y_min" value="-1" />
    <param name="global_planner/world_x_max" value="35" />
    <param name="global_planner/world_y_max" value="80" />
    <param name="global_planner/grid_resolution" value="0.5" />
    <param name="global_planner/box_resolution" value="0.3" />
    <param name="global_planner/box_max_size" value="10" />


    <param name="local_planner/horizon" value="5" />
    <param name="local_planner/obstacle_nomial_radius" value="0.4" />

    <param name="initial_goal/x" value="20" />
    <param name="initial_goal/y" value="68" />

    <param name="predictor/observation_queue" value="4" />
    <param name="predictor/ref_height" value="0.4" />
  </node>

  <node pkg="atypical_driving" type="integration_client" name="atypical_driving_client" output="screen">
  </node>


  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find atypical_driving)/launch/rviz_config/config.rviz" output="log" />
  
<!-- Nodes End -->
</launch>
